src/components/BillingRemaining.tsx

```diff
@@ -0,0 +1,157 @@
+import { IS_NIGHTLY_OR_DEBUG } from '@src/routes/utils'
+import { Spinner } from '@src/components/Spinner'
+import { useSelector } from '@xstate/react'
+import { useState } from 'react'
+
+import { CustomIcon } from '@src/components/CustomIcon'
+import type { BillingActor } from '@src/machines/billingMachine'
+
+export enum BillingRemainingMode {
+  ProgressBarFixed,
+  ProgressBarStretch,
+}
+
+export interface BillingRemainingProps {
+  mode: BillingRemainingMode
+  billingActor: BillingActor
+}
+
+const Error = (props: { error: Error }) => {
+  const [showMessage, setShowMessage] = useState(false)
+
+  const fadedBg = 'rgba(127, 127, 127, 0.2)'
+  const fadedFg = 'rgba(255, 255, 255, 0.6)'
+  const colors = {
+    color: fadedFg,
+    stroke: fadedFg,
+    fill: fadedFg,
+    backgroundColor: fadedBg,
+  }
+
+  return (
+    <div
+      onMouseEnter={() => setShowMessage(true)}
+      onMouseLeave={() => setShowMessage(false)}
+    >
+      {showMessage && (
+        <div
+          style={{
+            position: 'absolute',
+          }}
+        >
+          <div
+            className="rounded p-1"
+            style={{ ...colors, position: 'relative', top: -32 }}
+          >
+            {props.error.toString()}
+          </div>
+        </div>
+      )}
+      <div className="rounded" style={colors}>
+        <CustomIcon name="exclamationMark" className="w-5 h-5" />
+      </div>
+    </div>
+  )
+}
+
+const ProgressBar = (props: { max: number; value: number }) => {
+  const ratio = props.value / props.max
+
+  return (
+    <div
+      className="rounded w-full border-ml-black bg-ml-black"
+      style={{
+        height: 6,
+        borderWidth: '1px',
+        borderStyle: 'solid',
+      }}
+    >
+      <div
+        data-testid="billing-remaining-progress-bar-inner"
+        className="bg-ml-green"
+        style={{
+          width: Math.min(100, ratio * 100).toFixed(2) + '%',
+          height: '100%',
+          borderTopRightRadius: 4,
+          borderBottomRightRadius: 4,
+        }}
+      ></div>
+    </div>
+  )
+}
+
+const BillingCredit = (props: { amount: number }) => {
+  return props.amount === Infinity ? (
+    <CustomIcon data-testid="infinity" name="infinity" className="w-5 h-5" />
+  ) : Number.isNaN(props.amount) || props.amount === undefined ? (
+    <Spinner className="text-inherit w-4 h-4" />
+  ) : (
+    Math.max(0, Math.trunc(props.amount))
+  )
+}
+
+export const BillingRemaining = (props: BillingRemainingProps) => {
+  const billingContext = useSelector(
+    props.billingActor,
+    ({ context }) => context
+  )
+
+  const isFlex = props.mode === BillingRemainingMode.ProgressBarStretch
+  const cssWrapper = [
+    'bg-ml-green',
+    'select-none',
+    'cursor-pointer',
+    'py-1',
+    'rounded',
+    '!no-underline',
+    'font-semibold',
```

Comment:
```suggestion
```

I've been removing all bold-like fonts anywhere but like the onboarding headings (https://github.com/KittyCAD/modeling-app/pull/6373), I'm trying to only direct attention with scale, spacing, and color of text.